Coffee3.4/src/CoffeeBrush.java

```diff
@@ -0,0 +1,183 @@
+            if (water >= c_water && milk >= c_milk && coffee >= c_coffee) {
+                System.out.println("I have enough resources, making you a coffee!");
+                water = water - c_water;
+                milk = milk - c_milk;
+                coffee = coffee - c_coffee;
+                --cups;
+                money = money + c_cost;
+            } else if (water < c_water) {
+                System.out.println("Sorry, not enough water!");
+            } else if (milk < c_milk) {
+                System.out.println("Sorry, not enough milk!");
+            } else {
+                System.out.println("Sorry, not enough coffee!");
+            }
+        }
+        else {
+            System.out.println("Invalid action!");
+        }
+    }
+
+    /*remaining method*/
+    private void remaining(){
+        System.out.println("The coffee machine has:");
+        System.out.println(water + " of water");
+        System.out.println(milk + " of milk");
+        System.out.println(coffee + " of coffee beans");
+        System.out.println(cups + " of disposable cups");
+        System.out.println(money + "$ of money");
+    }
+
+    /*take method*/
+    private void take() {
+        System.out.println("I gave you $" + money);
+        money = money - money;
+    }
+
+    /*fill method*/
+    private void fill(){
+        Scanner f_scan = new Scanner(System.in);
+        System.out.println("Write how many ml of water do you want to add:");
+        int fill_water = f_scan.nextInt();
+        System.out.println("Write how many ml of milk do you want to add:");
+        int fill_milk = f_scan.nextInt();
+        System.out.println("Write how many grams of coffee beans do you want to add:");
+        int fill_coffee = f_scan.nextInt();
+        System.out.println("Write how many disposable cups of coffee do you want to add:");
+        int fill_cups = f_scan.nextInt();
+
+        water = water + fill_water;
+        milk = milk + fill_milk;
+        coffee = coffee + fill_coffee;
+        cups = cups + fill_cups;
+    }
+
+    private void setReady(){
+        state = State.Ready;
+        System.out.println("Write action (buy, fill, take, remaining, exit): ");
+    }
+
+    /*whole action*/
+    private void action(String input){
+        if(input.equals("remaining")) {
+            remaining();
+            setReady();
+        }
+        else if(input.equals("buy")){
+            buy();
+            state = State.SelectCoffee;
+            setReady();
+        }
+        else if(input.equals("fill")){
+            fill();
+            state = State.Filling;
+            setReady();
+        }
+        else if(input.equals("take")){
+            take();
+            setReady();
+        }
+        else if(input.equals("exit")){
+            state = State.Off;
+        }
+
+
+    }
+    /*exit method*/
+    public boolean isOff (){
+        return state == State.Off;
+    }
+    public static void main(String args[]) {
```

Comment:
Can you move logic to separate class, and create object on main method

Reply:
Done :)